web-client/src/presenter/computeds/caseDetailHeaderHelper.ts

```diff
@@ -1,8 +1,46 @@
 import { Case } from '../../../../shared/src/business/entities/cases/Case';
 import { ClientApplicationContext } from '@web-client/applicationContext';
 import { Get } from 'cerebral';
+import {
+  PRACTICE_TYPE,
+  ROLES,
+} from '@shared/business/entities/EntityConstants';
+import { RawIrsPractitioner } from '@shared/business/entities/IrsPractitioner';
+import { RawPractitioner } from '@shared/business/entities/Practitioner';
+import { RawUser } from '@shared/business/entities/User';
 import { state } from '@web-client/presenter/app.cerebral';
 
+const isUserADojPractitioner = (
+  user: RawUser | RawPractitioner | RawIrsPractitioner,
+): boolean => {
+  if (user.role !== ROLES.irsPractitioner) return false;
+  const irsPractitioner: RawPractitioner = user as RawPractitioner;
+  if (irsPractitioner.practiceType !== PRACTICE_TYPE.DOJ) return false;
+  return true;
+};
+
+const shouldShowRepresentAPartButton = (
+  caseDetail: RawCase,
+  user: RawUser | RawPractitioner | RawIrsPractitioner,
+  caseHasRespondent: boolean,
+  isRepresentAPartyForm: boolean,
+  isCaseSealed: boolean,
+  isCurrentPageFilePetitionSuccess: boolean,
+): boolean => {
+  const isDojPractitioner = isUserADojPractitioner(user);
+
+  const hideRepresentAPartyToDojUser =
```

Comment:
I'm getting a little confused by this variable name, but it may just be a me thing -- when `canDojPractitionerRepresentParty` is `true`, we set this 'hideRepresentParty' thing to `true`. Then, when `hide...` is true (as long as all the others eval to true), the function `shouldShow...()` returns `true`? If I'm following, would it be clearer to name this `show...` instead of `hide...`, or am I misreading this?

Reply:
You are correct, will update name. (Think I started implementing it as when to hide and then implemented when to show)